### ISSUE TITLE
Support Connecting to Multiple Systems with Event Hub Stream Binder

### ISSUE THREAD

**Issue by jd1** [Microsoft/spring-cloud-azure#508]
## Expected Behavior
Spring Cloud Stream supports connecting to multiple systems. An example of RabbitMQ can be found in the documentation: https://cloud.spring.io/spring-cloud-stream/spring-cloud-stream.html#multiple-systems
I would like to connect to multiple EventHub Namespaces.

## Current Behavior
The application fails to start with the exception: `org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type 'com.microsoft.azure.spring.integration.eventhub.api.EventHubOperation' available`

## Steps to Reproduce (for bugs)
* Create a project with spring-cloud-azure-eventhubs-stream-binder
* Set the following properties

```
spring.cloud.azure.eventhub.enabled=true
spring.cloud.azure.eventhub.connection-string=Endpoint=sb://...
spring.cloud.azure.eventhub.checkpointstorageaccount=eventstorage
spring.cloud.azure.eventhub.checkpointaccesskey=...

spring.cloud.stream.binders.azure1.type=eventhub
spring.cloud.stream.binders.azure1.environment.spring.cloud.azure.eventhub.connection-string=Endpoint=sb://...

spring.cloud.stream.binders.azure2.type=eventhub
spring.cloud.stream.binders.azure2.environment.spring.cloud.azure.eventhub.connection-string=Endpoint=sb://...

spring.cloud.stream.bindings.event1.destination=event1
spring.cloud.stream.bindings.event1.binder=azure1
spring.cloud.stream.bindings.event2.destination=event2
spring.cloud.stream.bindings.event2.binder=azure2
```

I also tried different combinations of properties:
* Without the property `spring.cloud.azure.eventhub.connection-string`, I get the following exception: `java.lang.IllegalArgumentException: Either 'spring.cloud.azure.eventhub.namespace' or 'spring.cloud.azure.eventhub.connection-string'`
* Without the properties `spring.cloud.azure.eventhub.connection-string` and `spring.cloud.azure.eventhub.enabled=true`, I get the following error:
```
Parameter 1 of method eventHubBinder in com.microsoft.azure.eventhub.stream.binder.config.EventHubBinderConfiguration required a bean of type 'com.microsoft.azure.spring.integration.eventhub.api.EventHubOperation' that could not be found.

The following candidates were found but could not be injected:
	- Bean method 'eventHubOperation' in 'AzureEventHubAutoConfiguration' not loaded because @ConditionalOnProperty (spring.cloud.azure.eventhub.enabled) found different value in property 'spring.cloud.azure.eventhub.enabled'
```

## Branch
* 1.1.0.RC4

## Your Environment
* Version used: 1.1.0.RC4
* Operating System and version (desktop or mobile): 1.1.0.RC4

**Comment by warrenzhu25** [comment]
Thanks for reporting this. We're looking into it.

**Comment by warrenzhu25** [comment]
Sample properties as below. Currently, the limitation is default binder must be provided. You can verify this on our daily snapshot build or built by yourself:

```
# Default binder
spring.cloud.azure.eventhub.connection-string=[]
spring.cloud.azure.eventhub.checkpoint-storage-account=[]
spring.cloud.azure.eventhub.checkpoint-access-key=[]

# Leaving this empty will use the default binder
# spring.cloud.stream.bindings.input.binder=azure1

# Another binder for azure1
spring.cloud.stream.binders.azure1.type=eventhub
spring.cloud.stream.binders.azure1.environment.spring.cloud.azure.eventhub.connection-string=[]
spring.cloud.stream.binders.azure1.environment.spring.cloud.azure.eventhub.checkpoint-storage-account=[]
spring.cloud.stream.binders.azure1.spring.cloud.azure.eventhub.checkpoint-access-key=[]

spring.cloud.stream.bindings.input.destination=eventhub1
spring.cloud.stream.bindings.input.binder=azure1
spring.cloud.stream.bindings.output.destination=eventhub1
spring.cloud.stream.bindings.output.binder=azure1
```

**Comment by vnicers** [comment]
@warrenzhu25 Can Service Bus Queue, Topic Stream Binder support it?

**Comment by warrenzhu25** [comment]
@vnicers Both Service Bus Stream binders don't support this now. But the fix should be easy and similar as 449a19f. If this feature is important for you, welcome to contribute since we have limited efforts on this recently.

**Comment by chenghaoharvey** [comment]
HI @warrenzhu25 , i have fixed this problem and contribute. Could you please help me to review the codes and merge it?[#516](https://github.com/microsoft/spring-cloud-azure/pull/516)

**Comment by ignesious** [comment]
@jd1  @chenghaoharvey  @warrenzhu25 @vnicers @edburns 
I need help with connecting to multiple eventhub via binders. 

MySpecification:-

Set of input streams should take from eventhub 1 and other set via eventhub 2.

 Binder would be apt for this usecase however finding issue on how to configure in application.properties.

Please help

**Comment by ignesious** [comment]
> ## Expected Behavior
> Spring Cloud Stream supports connecting to multiple systems. An example of RabbitMQ can be found in the documentation: https://cloud.spring.io/spring-cloud-stream/spring-cloud-stream.html#multiple-systems
> I would like to connect to multiple EventHub Namespaces.
> 
> ## Current Behavior
> The application fails to start with the exception: `org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type 'com.microsoft.azure.spring.integration.eventhub.api.EventHubOperation' available`
> 
> ## Steps to Reproduce (for bugs)
> * Create a project with spring-cloud-azure-eventhubs-stream-binder
> * Set the following properties
> 
> ```
> spring.cloud.azure.eventhub.enabled=true
> spring.cloud.azure.eventhub.connection-string=Endpoint=sb://...
> spring.cloud.azure.eventhub.checkpointstorageaccount=eventstorage
> spring.cloud.azure.eventhub.checkpointaccesskey=...
> 
> spring.cloud.stream.binders.azure1.type=eventhub
> spring.cloud.stream.binders.azure1.environment.spring.cloud.azure.eventhub.connection-string=Endpoint=sb://...
> 
> spring.cloud.stream.binders.azure2.type=eventhub
> spring.cloud.stream.binders.azure2.environment.spring.cloud.azure.eventhub.connection-string=Endpoint=sb://...
> 
> spring.cloud.stream.bindings.event1.destination=event1
> spring.cloud.stream.bindings.event1.binder=azure1
> spring.cloud.stream.bindings.event2.destination=event2
> spring.cloud.stream.bindings.event2.binder=azure2
> ```
> 
> I also tried different combinations of properties:
> 
> * Without the property `spring.cloud.azure.eventhub.connection-string`, I get the following exception: `java.lang.IllegalArgumentException: Either 'spring.cloud.azure.eventhub.namespace' or 'spring.cloud.azure.eventhub.connection-string'`
> * Without the properties `spring.cloud.azure.eventhub.connection-string` and `spring.cloud.azure.eventhub.enabled=true`, I get the following error:
> 
> ```
> Parameter 1 of method eventHubBinder in com.microsoft.azure.eventhub.stream.binder.config.EventHubBinderConfiguration required a bean of type 'com.microsoft.azure.spring.integration.eventhub.api.EventHubOperation' that could not be found.
> 
> The following candidates were found but could not be injected:
> 	- Bean method 'eventHubOperation' in 'AzureEventHubAutoConfiguration' not loaded because @ConditionalOnProperty (spring.cloud.azure.eventhub.enabled) found different value in property 'spring.cloud.azure.eventhub.enabled'
> ```
> 
> ## Branch
> * 1.1.0.RC4
> 
> ## Your Environment
> * Version used: 1.1.0.RC4
> * Operating System and version (desktop or mobile): 1.1.0.RC4

If you had solved this issue could you please give me the configuration on how you used ?

**Comment by ignesious** [comment]
@jd1  Please give me the git link where u have implemented this feature. I would help me to build my application.

**Comment by warrenzhu25** [comment]
As I know, this feature is not available in 1.1.0.RC4. You can try this feature on our daily snapshot build.

**Comment by saragluna** [comment]
@ignesious Please check the sample [here](https://github.com/microsoft/spring-cloud-azure/tree/master/spring-cloud-azure-samples/eventhubs-multibinders-sample).
